rom/laghos_rom.hpp

```diff
@@ -259,6 +266,204 @@ static double* getGreedyParam(ROM_Options& romOptions, const char* greedyParam)
     return iter->second;
 }
 
+class DMD_Sampler
```

Comment:
Classes `DMD_Sampler` and `ROM_Sampler` have a lot of repeated code in common, suggesting they should be derived from a common base class. There are also significant differences, so it would take some effort to rewrite with derived classes. @siuwuncheung do you think it would be too difficult or not worth the effort to make that change? The advantages would be reducing lines of code and avoiding duplication of existing code and future changes.

Reply:
I compare the two classes and get the impression that there are more difference than similarity at the current stage. Taking samples and recording the initial states work the same way. The `ROM_Sampler` has a lot of variables and options that are not useful in DMD, e.g. generator for nonlinear term basis, spacetime, SNS. Other variables are related to parametric cases, and may be re-used. Perhaps we can consider whether or not to merge the two sampler classes and rewrite with derived classes when we come to parametric DMD.